packages/shared/src/hooks/useOnPostClick.ts

```diff
@@ -48,6 +95,11 @@ export default function useOnPostClick({
   const { trackEvent } = useContext(AnalyticsContext);
   const { incrementReadingRank } = useIncrementReadingRank();
   const { queryKey: feedQueryKey, items } = useContext(ActiveFeedContext);
+  const { shouldUseFeedLayoutV1 } = useFeedLayout({
+    // need to provide something, otherwise it's always false
+    feedName: SharedFeedPage.MyFeed,
+  });
```

Comment:
The problem is, that we don't have the current feed in context on the post page. Same with the `feedName` prop, it's not sed on the post page. This is a poor attempt to get the feature flag value.

If you have better idea on how to get the feed name in there, please let me know. I wanted to avoid passing a query parameter to the URL (seems ugly).

Reply:
The hook does have `feedName` prop so I would look into passing it somehow. `useFeedLayout` is mostly here to contain the logic of checking the feeds that should show v1.  But bottom line we just need to check if `useFeature(feature.feedLayout) === FeedLayout.V1` for post page and since we don't care which feed it is on that page. So I would just evaluate `useFeature` here. Would that work?

Reply:
Yup if we don't care for this condition what feed we are one we should be good using useFeature, we've done so in other parts of the code changes

Reply:
I refactored out a new hook called `useFeatureFeedLayoutV1` and use that in both the `useFeedLayout` and `useOnPostClick` now. Should make the code more readable and easier to understand.